Segmentation fault when running tilepack with valid MBTiles config

Generating MBTiles crashes at the end with a segmentation fault. 

Here's the log and error output:

```
2025/05/27 17:47:03 Expecting to fetch 9700 tiles
2025/05/27 17:47:03 Created mbtiles output
  79% |██████████████████████████████████████████████████████████████████████████████████████████████████████████                             | (150 tile/s) [39s:13s]2025/05/27 17:47:46 Job queue closed
 100% |████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| (193 tile/s)2025/05/27 17:47:56 Finished making tile requests
2025/05/27 17:47:56 Processed 9700 tiles
2025/05/27 17:47:56 Finished processing tiles
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x7413a6]

goroutine 1 [running]:
database/sql.(*Tx).grabConn(0x0, {0xe340c8, 0x12d71c0})
        database/sql/sql.go:2246 +0x46
database/sql.(*Tx).ExecContext(0x0, {0xe340c8, 0x12d71c0}, {0xc72d43, 0x3a}, {0xc0009359e0, 0x2, 0x2})
        database/sql/sql.go:2504 +0x4b
database/sql.(*Tx).Exec(...)
        database/sql/sql.go:2517
github.com/tilezen/go-tilepacks/tilepack.(*mbtilesOutputter).Close(0xc000135740)
        github.com/tilezen/go-tilepacks/tilepack/mbtiles_outputter.go:41 +0x159
main.main()
        ./main.go:316 +0x20ac
```

Command Used:
```

tilepack -dsn /tmp/3_googletiles.mbtiles \
                      -url-template "https://mt0.google.com/vt?lyrs=s&x={x}&y={y}&z={z}" \
                      -bounds 26.905668,88.065078,26.961207,88.118987 \
                      -zooms 12-19 \
                      -output-mode mbtiles \
                      -mbtiles-format jpg \
                      -ensure-gzip=false \
                      -tileset-name fmtm_3_googletiles
```

--- a/tilepack/mbtiles_outputter.go
+++ b/tilepack/mbtiles_outputter.go
@@ -35,6 +35,20 @@ type mbtilesOutputter struct {
 func (o *mbtilesOutputter) Close() error {
 	var err error
 
+	// Commit remaining tile transaction if it exists
+	if o.txn != nil {
+		if err = o.txn.Commit(); err != nil {
+			return fmt.Errorf("failed to commit final tile batch: %w", err)
+		}
+		o.txn = nil
+	}
+
+	// Start a new transaction for metadata
+	o.txn, err = o.db.Begin()
+	if err != nil {
+		return fmt.Errorf("failed to begin metadata transaction: %w", err)
+	}
+
 	// Save the metadata
 	for name, value := range o.metadata.metadata {
 		q := "INSERT OR REPLACE INTO metadata (name, value) VALUES(?, ?)"
@@ -163,6 +177,8 @@ func (o *mbtilesOutputter) Save(tile maptile.Tile, data []byte) error {
 			return err
 		}
 		o.batchCount = 0
+		// NOTE if the total tiles to download are a multiple of batchSize,
+		// this will run and set o.txn to nil prior to .Close() being called
 		o.txn = nil
 	}
 

